src/lib/models/subscriptions.rs

```diff
@@ -55,6 +58,65 @@ impl PartialEq for Subscription {
 }
 impl Eq for Subscription {}
 
+impl UpdateStatus for Subscription {
+    fn get_raw_status(&self) -> Option<String> {
+        self.status.clone()
+    }
+
+    fn get_raw_status_history(&self) -> Option<JsonValue> {
+        self.status_history.clone()
+    }
+
+    fn set_raw_status(&mut self, v: Option<String>) {
+        self.status = v;
+    }
+
+    fn set_raw_status_history(&mut self, v: Option<JsonValue>) {
+        self.status_history = v;
+    }
+}
+
+impl Subscription {
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I'm in two minds about whether creating a new struct with the public fields of Subscription is worth it. 

Using a struct is much nicer so that all your arguments are properly named. 

But we're trading off adding another struct with the fact that we only actually have one constructor....I'm not sure it's worth it given that it's a small project that's well tested.